Library/ViewModels/RewardsCollectionViewModel.swift

```diff
@@ -215,6 +276,19 @@ private func shouldNavigateToReward(project: Project, reward: Reward, refTag _:
   project.state == .live && (!userIsBacking(reward: reward, inProject: project) || reward.hasAddOns)
 }
 
+private func shouldTriggerEditRewardPrompt(_ data: PledgeViewData) -> Bool {
+  // If the user is not backing the project then there is no need to show the prompt.
+  guard
+    userIsBackingProject(data.project),
+    let backing = data.project.personalization.backing
+  else { return false }
+
+  let rewardUnchanged = data.rewards.first?.id == backing.reward?.id
+
+  // We show the prompt if they have previously backed with add-ons and they selecting a new reward.
+  return backing.addOns?.isEmpty == false && !rewardUnchanged
```

Comment:
It took me a second to see what this was doing, consider:
`let rewardChanged = data.rewards.first?.id != backing.reward?.id`

and then
```suggestion
  return !backing.addOns?.isEmpty && rewardChanged
```

Reply:
Ah thanks that's clearer, I still need to do the `== false` because the boolean is optional.